docs/src/whatsnew/latest.rst

```diff
@@ -96,6 +96,8 @@ This document explains the changes made to Iris for this release
    lazy data from file. This will also speed up coordinate comparison.
    (:pull:`5610`)
 
+#. `@rcomer`_ modified :func:`~iris.analysis.stats.pearsonr` so it preserves
+   lazy data in all cases and also runs a little faster.  (:pull:`5638`)
```

Comment:
Could we have a benchmark to demonstrate this?

Reply:
I would probably need some help with that as my last attempt to run the benchmarks locally failed.  See you at next surgery?

Reply:
First attempt now added.